spec/hydra/works/models/collection_spec.rb

```diff
@@ -1,229 +1,87 @@
+  let(:collection) { described_class.new }
 
   let(:collection1) { described_class.new }
-  let(:collection2) { described_class.new }
-  let(:collection3) { described_class.new }
-  let(:collection4) { described_class.new }
-
   let(:generic_work1) { Hydra::Works::GenericWork.new }
-  let(:generic_work2) { Hydra::Works::GenericWork.new }
-  let(:generic_work3) { Hydra::Works::GenericWork.new }
-  let(:generic_work4) { Hydra::Works::GenericWork.new }
 
   describe '#collections' do
     it 'returns empty array when only works are aggregated' do
-      subject.ordered_members << generic_work1
-      subject.ordered_members << generic_work2
-      expect(subject.collections).to eq []
+      collection.ordered_members << generic_work1
+      expect(collection.collections).to eq []
     end
 
     context 'with other collections & works' do
+      let(:collection2) { described_class.new }
       before do
-        subject.ordered_members << collection1
-        subject.ordered_members << collection2
-        subject.ordered_members << generic_work1
-        subject.ordered_members << generic_work2
+        collection.ordered_members << collection1
+        collection.ordered_members << collection2
+        collection.ordered_members << generic_work1
       end
 
       it 'returns only collections' do
-        expect(subject.ordered_collections).to eq [collection1, collection2]
+        expect(collection.ordered_collections).to eq [collection1, collection2]
       end
     end
   end
 
   describe '#works' do
-    it 'returns empty array when only collections are aggregated' do
-      subject.ordered_members << collection1
-      subject.ordered_members << collection2
-      expect(subject.ordered_works). to eq []
+    subject { collection.works }
+    context "when only collections are aggregated" do
+      it 'returns empty array when only collections are aggregated' do
+        collection.ordered_members << collection1
+        expect(subject).to eq []
+      end
     end
 
     context 'with collections and generic works' do
+      let(:generic_work2) { Hydra::Works::GenericWork.new }
       before do
-        subject.ordered_members << collection1
-        subject.ordered_members << collection2
-        subject.ordered_members << generic_work1
-        subject.ordered_members << generic_work2
+        collection.ordered_members << collection1
+        collection.ordered_members << generic_work1
+        collection.ordered_members << generic_work2
       end
 
       it 'returns only generic works' do
-        expect(subject.ordered_works). to eq [generic_work1, generic_work2]
+        expect(subject).to eq [generic_work1, generic_work2]
       end
     end
   end
 
-  describe "#ordered_work_ids" do
-    it "returns IDs of ordered works" do
-      subject.ordered_members << generic_work1
-      expect(subject.ordered_work_ids).to eq [generic_work1.id]
+  describe '#ordered_works' do
+    subject { collection.works }
```

Comment:
I may be misreading this, but this block looks identical to the code above on lines 30-35.  Should the subject here be `collection.ordered_works` instead?